src/ansys/fluent/core/utils/search.py

```diff
@@ -212,10 +243,230 @@ def inner(tree, path, root_path):
                     next_path = f'{path}.{k}["<name>"]'
                 else:
                     next_path = f"{path}.{k}"
+                type_ = "Object" if isinstance(v, Mapping) else v
+                api_object_names.add(k)
+                if "tui" in next_path:
+                    api_tui_objects.append(f"{next_path} ({type_})")
+                else:
+                    api_objects.append(f"{next_path} ({type_})")
                 if _match(k, word, match_whole_word, match_case):
-                    type_ = "Object" if isinstance(v, Mapping) else v
                     print(f"{next_path} ({type_})")
             if isinstance(v, Mapping):
                 inner(v, next_path, root_path)
 
     inner(api_tree, "", root_path)
+
+    def _write_api_tree_file(
+        api_objects: list,
+        name: Optional[bool] = None,
+        text: Optional[bool] = False,
+        tui: Optional[bool] = False,
+    ):
+        api_tree_file = get_api_tree_file_name(text=text, name=name, tui=tui)
+        api_tree_file.touch()
+        with open(api_tree_file, "w") as file:
+            for api_object in api_objects:
+                file.write(f"{api_object}")
+                file.write("\n")
+
+    if pyfluent.WRITE_API_SEARCH_OBJECTS_FILE:
+
+        text_file_folder = Path(os.path.join(_THIS_DIRNAME, "api_tree"))
+        text_file_folder.mkdir(parents=True, exist_ok=True)
+
+        _write_api_tree_file(api_objects=api_objects, text=True)
+        _write_api_tree_file(api_objects=api_tui_objects, tui=True)
+        _write_api_tree_file(api_objects=api_object_names, name=True)
```

Comment:
Are these files based on the api_tree data, independent of search-strings? Can we package them with PyFluent? Note that, user may not have write permission to `_THIS_DIRNAME`.

Reply:
@mkundu1 

Are these files based on the api_tree data, independent of search-strings?  - Yes

Can we package them with PyFluent? - Yes that's why we have created `api_tree` folder inside `ansy/fluent/core/utils` dir so that we can package them with PyFluent.

Note that, user may not have write permission to _THIS_DIRNAME. - Yes, I agree.

Reply:
Use won't need to generate them.

Reply:
How are we updating these files? E.g., if settings API is changed, will these files be updated during codegen?

I also don't think these files should be version-controlled.

Reply:
I think we should support latest release version only (24R2 for now) otherwise there will be total 15 files in `api_tree` folder. (3 for each version)

To update these files during codegen we will have to execute following code in allapigen.py.

```
if __name__ == "__main__":
  import ansys.fluent.core as pyfluent
  from ansys.fluent.core.utils.search import _seach
  pyfluent.WRITE_API_SEARCH_OBJECTS_FILE = True
  _search("", version="242")
```

Reply:
I'm fine if we support this only for 24.2 in the first merge.

Reply:
We have added this to `allapigen.py`.